overlord/snapstate/autorefresh.go

```diff
@@ -641,8 +641,12 @@ func inhibitRefresh(st *state.State, snapst *SnapState, info *snap.Info, checker
 		checkerErr = nil
 	}
 
-	// Send the notification asynchronously to avoid holding the state lock.
-	asyncPendingRefreshNotification(context.TODO(), userclient.New(), refreshInfo)
+	// if the refresh is manual the error message already carries the information so don't notify
+	if snapst.IsAutoRefresh {
```

Comment:
I'm confused, how does IsAutoRefresh ends up set in SnapState to use here, it's an ephemeral state, it seems more of a bug that is there at all? Afaict this is the only place that would read it out of there anyway so far. The other gets it from flags or SnapSetup. Link handlers copy some things from SnapSetup to SnapState but not IsAutoRefresh afaict.

Reply:
Indeed, that was a mistake. I changed it to take in SnapSetup now